src/js/angular/security/controllers.js

```diff
@@ -355,32 +366,38 @@ securityCtrl.controller('CommonUserCtrl', ['$rootScope', '$scope', '$http', 'toa
             setGrantedAuthorities($scope);
         };
 
-        $scope.hasReadPermission = function (repositoryId) {
+        $scope.hasReadPermission = function (repository) {
+            let uniqueKey = createUniqueKey(repository);
             return $scope.userType === UserType.ADMIN || $scope.userType === UserType.REPO_MANAGER
-                || $scope.grantedAuthorities.READ_REPO[repositoryId]
-                || $scope.grantedAuthorities.WRITE_REPO[repositoryId]
-                || repositoryId !== SYSTEM_REPO
-                && ($scope.grantedAuthorities.READ_REPO['*'] || $scope.grantedAuthorities.WRITE_REPO['*']);
+                || repository.id !== SYSTEM_REPO
+                && ($scope.grantedAuthorities.READ_REPO['*'] || $scope.grantedAuthorities.WRITE_REPO['*'])
+                || $scope.grantedAuthorities.READ_REPO[uniqueKey]
+                || $scope.grantedAuthorities.WRITE_REPO[uniqueKey];
         };
 
-        $scope.hasWritePermission = function (repositoryId) {
+        $scope.hasWritePermission = function (repository) {
+            let uniqueKey = createUniqueKey(repository);
             return $scope.userType === UserType.ADMIN || $scope.userType === UserType.REPO_MANAGER
-                || $scope.grantedAuthorities.WRITE_REPO[repositoryId]
-                || repositoryId !== SYSTEM_REPO && $scope.grantedAuthorities.WRITE_REPO['*'];
+                || repository.id !== SYSTEM_REPO && $scope.grantedAuthorities.WRITE_REPO['*']
+                || $scope.grantedAuthorities.WRITE_REPO[uniqueKey];
         };
 
-        $scope.readCheckDisabled = function (repositoryId) {
-            return $scope.hasWritePermission(repositoryId)
-                || repositoryId !== SYSTEM_REPO && repositoryId !== '*' && $scope.grantedAuthorities.READ_REPO['*']
+        $scope.readCheckDisabled = function (repository) {
+            return $scope.hasWritePermission(repository)
+                || repository.id !== SYSTEM_REPO && repository !== '*' && $scope.grantedAuthorities.READ_REPO['*']
```

Comment:
`repository !== '*'` is not correct. I think it should be `repository.id !== '*'`. Can you double-check this. Same in below function on line 393.